src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksTaxRuleDependencyServiceImpl.java

```diff
@@ -0,0 +1,261 @@
+package edu.cornell.kfs.pmw.batch.service.impl;
+
+import java.sql.Date;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.vnd.VendorConstants;
+import org.kuali.kfs.vnd.businessobject.VendorHeader;
+import org.kuali.rice.core.api.config.property.ConfigurationService;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+
+import edu.cornell.kfs.pmw.batch.PaymentWorksConstants;
+import edu.cornell.kfs.pmw.batch.PaymentWorksKeyConstants;
+import edu.cornell.kfs.pmw.batch.TaxRule;
+import edu.cornell.kfs.pmw.batch.businessobject.KfsVendorDataWrapper;
+import edu.cornell.kfs.pmw.batch.businessobject.PaymentWorksIsoFipsCountryItem;
+import edu.cornell.kfs.pmw.batch.businessobject.PaymentWorksVendor;
+import edu.cornell.kfs.pmw.batch.service.PaymentWorksBatchUtilityService;
+import edu.cornell.kfs.pmw.batch.service.PaymentWorksTaxRuleDependencyService;
+import edu.cornell.kfs.sys.CUKFSConstants;
+
+public class PaymentWorksTaxRuleDependencyServiceImpl implements PaymentWorksTaxRuleDependencyService {
+    private static final Logger LOG = LogManager.getLogger(PaymentWorksTaxRuleDependencyServiceImpl.class);
+    
+    protected DateTimeService dateTimeService;
+    protected PaymentWorksBatchUtilityService paymentWorksBatchUtilityService;
+    protected ConfigurationService configurationService;
+    private DateTimeFormatter dateTimeFormatter;
+    
+
+    @Override
+    public KfsVendorDataWrapper populateTaxRuleDependentAttributes(PaymentWorksVendor pmwVendor,
```

Comment:
Is this in fact only populating the tax rule dependent attributes?

Reply:
Yes, I believe that is true ..... we do set the country code values, but this also used to derive the tax rule, so I think the naming makes sense.  Do you see something that isn't tax rule dependent?

Reply:
I see what you mean, I think the name makes sense then.
I think what makes it a bit confusing is that the populateTaxRuleDependentAttributes just calls populateVendorHeaderOnKfsVendorDataWrapper. I only found this vendorHeader.setVendorForeignIndicator(!isUnitedStatesFipsCountryCode(vendorFipsCountryCode)); that seems to populate a field that doesn't depend on tax rule.

Maybe it would have made sense to move this code: 
        String vendorFipsCountryCode = convertIsoCountryCodeToFipsCountryCode(pmwVendor.getRequestingCompanyTaxCountry(), paymentWorksIsoToFipsCountryMap);
        TaxRule taxRule = determineTaxRuleToUseForDataPopulation(pmwVendor, vendorFipsCountryCode);

        VendorHeader vendorHeader = vendorDataWrapper.getVendorDetail().getVendorHeader();
        vendorHeader.setVendorForeignIndicator(!isUnitedStatesFipsCountryCode(vendorFipsCountryCode)); 
to populateTaxRuleDependentAttributes and then pass the necessary values as input to populateVendorHeaderOnKfsVendorDataWrapper?

Reply:
That didn't turn out as smoothly as expected, but I reorganized the code, lets see if this makes more sense.